zerver/lib/ccache.py

```diff
@@ -54,7 +53,7 @@ def der_encode_tlv(tag, value):
 
 def der_encode_integer_value(val):
     # type: (int) -> bytes
-    if not isinstance(val, six.integer_types):
+    if not isinstance(val, int):
```

Comment:
I think this code was written earlier.  It doesn't have tests and is for a limited use case, so our strategy is to avoid making changes when possible :)

Reply:
`isinstance` should be fine, but there's a few places where you did something more complicated.  Can you separate those in another commit?  I agree the `isinstance` replacements should be safe.

Reply:
I provide rationale for each of the things. Is this sufficient.